### ISSUE TITLE
Auto-layout when creating nodes

### ISSUE THREAD

**Issue by MichaelMauderer** [enso-org/ide#1550]
### Summary
New nodes should be placed, so they don't overlap existing nodes. 

### Value
Better overview over the graph while editing it, and less manual moving nodes around required. 

### Specification
When creating a new node, the new node should be placed at an empty spot in the graph that is 
* close to the source of the creation (i.e., where the user initiated the node creation action, which is where we currently place the node) and 
* in a spot that has enough space to contain the node.

The new size of the node should be estimated by some basic assumption, if they are known: standard node size (just come constant that we guesstimate), known variable names of the new node(if there is some input already defined through the previously selected node). 
We do not relocate a node that was newly created but gets too large, e.g., due to default arguments. 

### Acceptance Criteria & Test Cases
Scenario showing the node created in the described way.
Test cases that assert a new created node does not overlap other nodes (under the given size assumptions).

**Comment by s9ferech** [comment]
At what exact moment should the node be positioned? When it is initially created with an empty content, or when the user has entered some initial text and hits enter for the first time? @wdanilo

**Comment by wdanilo** [comment]
@s9ferech When we press tab the searcher should already be positioned in the "right" position. The "right" position means the Y-position as it is now and the X position on the right in the first place that is free for a hardcoded width. So the algorithm after pressing tab should be:
1. Move down A pixels (A is a value, like current offset in the product)
2. Assume the node width required is B, move right as far as needed to find a spot free for a node of B width
3. Allow editing of the node. Even if the node width will be bigger than B, keep it at that position.

**Comment by s9ferech** [comment]
The acceptance criteria contain "A scenario showing the node created in the described way." What exactly is meant by that?

**Comment by MichaelMauderer** [comment]
> The acceptance criteria contain "A scenario showing the node created in the described way." What exactly is meant by that?

Show a video of the working feature, and create a description for our test scenario to verify it is working correctly.

**Comment by s9ferech** [comment]
I don't think it will be possible to create automated tests for this without larger refactorings. What do you think @MichaelMauderer?

**Comment by MichaelMauderer** [comment]
Agree that automated tests at the moment are beyond scope. But can you try to set up a demo scene that initializes nodes that would usually overlap but now get auto-layouted? That should be feasible, and we can check that demo scene as part of the test scenario.

**Comment by s9ferech** [comment]
I added some nodes to the "interface" demo scene that are laid out automatically, as shown in the PR description. As part of the test scenario, one could simply check if those nodes are displayed without overlap and in the right order. Do you think that solution is good enough @MichaelMauderer?

**Comment by MichaelMauderer** [comment]
Yes, that sounds perfect.

**Comment by s9ferech** [comment]
Can you add this to the test scenario, @MichaelMauderer?

**Comment by MichaelMauderer** [comment]
> Can you add this to the test scenario, @MichaelMauderer?

Let's update the scenario when this is merged.
